bundle/src/main/java/com/adobe/acs/commons/forms/helpers/impl/PostRedirectGetWithCookiesFormHelperImpl.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    @Override
+    public final void sendRedirect(Form form, Page page, String formSelector, SlingHttpServletResponse response)
+            throws IOException, JSONException {
+        final String url = this.getRedirectPath(form, page, formSelector);
+        addFlashCookie(response, form);
+        response.sendRedirect(url);
+    }
+
+    @Override
+    public final void sendRedirect(Form form, Resource resource, String formSelector,
+            SlingHttpServletResponse response) throws IOException, JSONException {
+        final String url = this.getRedirectPath(form, resource, formSelector);
+        addFlashCookie(response, form);
+        response.sendRedirect(url);
+    }
+
+    @Override
+    protected final boolean doHandleGet(final String formName, final SlingHttpServletRequest request) {
+        //noinspection SimplifiableIfStatement
+        if (StringUtils.equalsIgnoreCase("GET", request.getMethod())) {
+            return (findCookie(request, this.getGetLookupKey(formName)) != null);
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    protected String getRawFormData(final String formName, final SlingHttpServletRequest request,
+            final SlingHttpServletResponse response) {
+        final Cookie cookie = findCookie(request, getGetLookupKey(formName));
+        if (response != null) {
+            cookie.setMaxAge(0);  // expire the cookie
+            response.addCookie(cookie);
+        } else {
+            log.warn("SlingHttpServletResponse required for removing cookie. Please use formHelper.getForm(\"" + formName + "\", slingRequest, slingResponse);");
+        }
+        // Get the QP lookup for this form
+        return this.decode(cookie.getValue());
+    }
+
+    @Override
+    protected final String getRedirectPath(final Form form, final String path, final String formSelector) throws
+            JSONException {
+        String redirectPath = path;
+        redirectPath += this.getSuffix();
+        if (StringUtils.isNotBlank(formSelector)) {
+            redirectPath += "/" + formSelector;
+        }
+        return redirectPath;
+    }
+
+    /**
+     * Encodes URL data, escaping characters such as "+" and "="
+     *
+     * @param unencoded
+     * @return
+     */
+    @Override
+    protected final String encode(String unencoded) {
+        return StringUtils.isBlank(unencoded) ? "" : org.apache.sling.commons.json.http.Cookie.escape(unencoded);
+    }
+
+    /**
+     * Decodes URL data.
+     *
+     * @param encoded
+     * @return
+     */
+    @Override
+    protected final String decode(String encoded) {
+        return StringUtils.isBlank(encoded) ? "" : org.apache.sling.commons.json.http.Cookie.unescape(encoded);
+    }
+
+    /**
+     * Find the cookie on the request
+     * @param request
+     * @param name
+     * @return form cookie or null
+     */
+    protected Cookie findCookie(SlingHttpServletRequest request, String name) {
```

Comment:
I dont think its worth changing at this point; but we have a handy CookieUtil  that does a lot of Cookie reading/writing